source/slang/ir-legalize-types.cpp

```diff
@@ -459,6 +466,8 @@ static LegalVal legalizeFieldAddress(
     IRStructKey*                fieldKey)
 {
     auto builder = context->builder;
+    if (type.flavor == LegalType::Flavor::none)
```

Comment:
Fixed another bug in `legalizeFieldAddress`, we need to check whether the field we are loading is an empty struct. If it is, just return None.